pkg/server/api/agent/v1/service_test.go

```diff
@@ -117,6 +117,93 @@ var (
 	}
 )
 
+func TestCountAgents(t *testing.T) {
+	ids := []spiffeid.ID{
+		spiffeid.Must("example.org", "node1"),
+		spiffeid.Must("example.org", "node2"),
+		spiffeid.Must("example.org", "node3"),
+	}
+
+	for _, tt := range []struct {
+		name       string
+		count      int32
+		code       codes.Code
+		dsError    error
+		err        string
+		expectLogs []spiretest.LogEntry
+	}{
+		{
+			name:  "0 nodes",
+			count: 0,
+		},
+		{
+			name:  "1 node",
+			count: 1,
+		},
+		{
+			name:  "2 nodes",
+			count: 2,
+		},
+		{
+			name:  "3 nodes",
+			count: 3,
+		},
+		{
+			name:    "ds error",
+			code:    codes.Internal,
+			dsError: errors.New("some error"),
+			err:     "failed to count agents: some error",
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Failed to count agents",
+					Data: logrus.Fields{
+						logrus.ErrorKey: "some error",
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test := setupServiceTest(t)
+			defer test.Cleanup()
+
+			for i := 0; i < int(tt.count); i++ {
+				_, err := test.ds.CreateAttestedNode(ctx, &datastore.CreateAttestedNodeRequest{
+					Node: &common.AttestedNode{
+						SpiffeId:            ids[i].String(),
+						AttestationDataType: "t1",
+						CertSerialNumber:    "badcafe",
+						CertNotAfter:        time.Now().Add(-time.Minute).Unix(),
+						NewCertNotAfter:     time.Now().Add(time.Minute).Unix(),
+						NewCertSerialNumber: "new badcafe",
+						Selectors: []*common.Selector{
+							{Type: "a", Value: "1"},
+							{Type: "b", Value: "2"},
+						},
+					},
+				})
+				require.NoError(t, err)
+			}
+
+			test.ds.SetNextError(tt.dsError)
+			resp, err := test.client.CountAgents(ctx, &agentpb.CountAgentsRequest{})
+
+			spiretest.AssertLogs(t, test.logHook.AllEntries(), tt.expectLogs)
+			if tt.err != "" {
+				spiretest.RequireGRPCStatusContains(t, err, tt.code, tt.err)
+				require.Nil(t, resp)
+				return
+			}
+
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+			require.Equal(t, resp.Count, tt.count)
```

Comment:
instead of only count, can you compare response protos? just to know we are getting response we want in all fields.
you can do that using `spiretest.AssertProtoEqual(t, tt.resp, resp`
at the same time, in golang require.Equal/assert.Equal expects (t, EXPECTED, ACTUAL)

This same comment is for all count RPCs